src/test/scala/chiselTests/util/PlusArgsValueSpec.scala

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chiselTests.util
+
+import chisel3._
+import chisel3.stage.{ChiselGeneratorAnnotation, ChiselStage}
+import chisel3.testers.BasicTester
+import chisel3.util.circt.PlusArgsValue
+
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import scala.io.Source
+
+private class PlusArgsValueTop extends Module {
+  val io = IO(new Bundle {
+    val wf = Output(UInt(1.W))
+    val wv = Output(UInt(32.W))
+    val xf = Output(UInt(1.W))
+    val xv = Output(UInt(32.W))
+  })
+  val tmpw = PlusArgsValue(UInt(32.W), "FOO=%d")
+  val tmpx = PlusArgsValue(io.xv, "BAR=%d")
```

Comment:
and if the format string is user controlled please show some tests/applications of things that are not %d

Reply:
To what end?  The string is passed by chisel verbatim to firrtl.  These are not end-to-end verilog generation tests.

Reply:
This is a user API we are designing. I am trying to understand the utility of exposing this string to the user, as well as the utility of allowing arbitrary gen data types. If we have no use for it, and users are only ever going to reasonably get UInts back, then let's remove the ambiguity int he API. If we are saying "sure, you can format your results as a float, or signed number" then what do we expect the users to assign those to?

Reply:
I guess we are trying to formalize https://github.com/chipsalliance/rocket-chip/blob/master/src/main/resources/vsrc/plusarg_reader.v, which does expose the format string, but not the output data type (which is hard-coded to a UInt of a user-specified width).

Reply:
We are not trying to formalize that.  We are trying to provide access to functionality described in SV 2012 section 21.6.  That that blackbox will be directly implementable is a (intended) side effect.

Reply:
A missing point is it is not the case that the format has to be "STR=FORMAT", it can be any pattern match.